extensions/aris/src/bigDataCluster/controller/controller.ts

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { EndpointRouterApi } from './apiGenerated';
+import { IEndPointsResponse, IControllerError, IEndPointsRequest, IHttpResponse, IEndPoint } from './types';
+
+export class ArisController {
+
+	private static adjustUrl(url: string): string {
+		if (!url) {
+			return undefined;
+		}
+
+		url = url.trim().replace(/ /g, '').replace(/,(\d+)$/, ':$1');
+		if (!url.includes('://')) {
+			url = `https://${url}`;
+		}
+		return url;
+	}
+
+	public static async getEndPoints(
```

Comment:
Please use `export function getEndpoints` instead of writing a whole class here. I'd suggest:
- Call this file `clusterController`
- Don't use class for static things, module w/ function exports is preferred so this becomes an exported function
- Usage elsewhere is to call `clusterController.getEndpoints`

For unit testability, you should accept an optional EndpointRouterApi as a parameter to the function and in testing have a mocked version instead.

Reply:
fixed.